test/functional/apps/discover/_runtime_fields_editor.ts

```diff
@@ -61,32 +61,35 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
     });
 
     it('allows creation of a new field', async function () {
-      await createRuntimeField('runtimefield');
-      await PageObjects.header.waitUntilLoadingHasFinished();
+      const field = '_runtimefield';
+      await createRuntimeField(field);
       await retry.waitForWithTimeout('fieldNames to include runtimefield', 5000, async () => {
         const fieldNames = await PageObjects.discover.getAllFieldNames();
-        return fieldNames.includes('runtimefield');
+        return fieldNames.includes(field);
       });
     });
 
     it('allows editing of a newly created field', async function () {
-      await PageObjects.discover.editField('runtimefield');
-      await fieldEditor.setName('runtimefield edited');
+      const field = '_runtimefield-before-edit';
+      await createRuntimeField(field);
+      const newFieldName = '_runtimefield-after-edit';
+      await PageObjects.discover.editField(field);
+      await fieldEditor.setName(newFieldName, true);
```

Comment:
@majagrubic the second argument to `setName` resolved the flakiness, since it made sure, that the field is cleared first. in the flaky version the new fieldname was appended to the existing one ... and the screens didn't tell .. but analyzing the HTML of the failure did.